src/town.h

```diff
@@ -219,6 +221,7 @@ DECLARE_ENUM_AS_BIT_SET(TownActions)
 
 extern const byte _town_action_costs[TACT_COUNT];
 extern TownID _new_town_id;
+extern TownID _num_of_shown_zones;
```

Comment:
I don't know if TownID or uint16 would be better here, as it does't refer to any town, but if in the future TownID becomes bigger, this should grow as well.

Reply:
It's a count, not a reference to one specific town, hence it's not a TownID. (Looking the value up in the towns array would not make logical sense.) Since it's a global count, not a member variable, I'd suggest using size_t for it since increase in the memory usage would be insignificant.

Reply:
This is definitely not a TownID.

Reply:
It's easy to see this is not a TownID: If you pass the value in to `Town::Get(_num_of_shown_zones)` the Town object returned (if any) has no meaning in context.

Reply:
I removed it entirely. After pushing this commit I had an idea, and reworking the whole thing (to use k-d trees).